src/ports/postgres/modules/knn/knn.sql_in

```diff
@@ -0,0 +1,440 @@
+	Argument 6: Name of column having feature vector Ids in test data table
+	Argument 7: Name of output table
+	Argument 8: c for classification task, r for regression task
+	Argument 9: value of k. Default will go as 1';
+    END IF;
+END;
+$$ LANGUAGE plpgsql VOLATILE
+m4_ifdef(`__HAS_FUNCTION_PROPERTIES__', `READS SQL DATA', `');
+
+CREATE OR REPLACE FUNCTION MADLIB_SCHEMA.knn(
+) RETURNS VOID AS $$
+BEGIN    
+    EXECUTE $sql$ select * from MADLIB_SCHEMA.knn('help') $sql$;
+END;
+$$ LANGUAGE plpgsql VOLATILE
+m4_ifdef(`__HAS_FUNCTION_PROPERTIES__', `READS SQL DATA', `');
+
+
+CREATE OR REPLACE FUNCTION MADLIB_SCHEMA.knn(
+    point_source VARCHAR,
+    point_column_name VARCHAR,
+    label_column_name VARCHAR,
+    test_source VARCHAR,
+    test_column_name VARCHAR,
+    id_column_name VARCHAR,
+    output_table VARCHAR,
+    operation VARCHAR,
+    k INTEGER
+) RETURNS VARCHAR AS $$
+DECLARE
+    class_test_source REGCLASS;
+    class_point_source REGCLASS;
+    l FLOAT;
+    outputTableFlag INTEGER;
+    id INTEGER;
+    vector DOUBLE PRECISION[];
+    cur_pid integer;
+    oldClientMinMessages VARCHAR;
+    returnstring VARCHAR;
+BEGIN
+    oldClientMinMessages :=
+        (SELECT setting FROM pg_settings WHERE name = 'client_min_messages');
+    EXECUTE 'SET client_min_messages TO warning';
+    PERFORM MADLIB_SCHEMA.__knn_validate_src(test_source);
+    PERFORM MADLIB_SCHEMA.__knn_validate_src(point_source);
+    class_test_source := test_source;
+    class_point_source := point_source;
+    --checks
+    IF (k <= 0) THEN
+        RAISE EXCEPTION 'KNN error: Number of neighbors k must be a positive integer.';
+    END IF;
+    IF (operation != 'c' AND operation != 'r') THEN
+        RAISE EXCEPTION 'KNN error: put r for regression OR c for classification.';
+    END IF;
+    PERFORM MADLIB_SCHEMA.create_schema_pg_temp();
+    
+    EXECUTE 
+	$sql$ 
+	SELECT count(*) FROM information_schema.tables WHERE table_name = '$sql$ || output_table || $sql$'$sql$ into outputTableFlag; 
+    IF (outputTableFlag != 0) THEN
+	RAISE Exception '% table exists. Drop it or use different name', output_table;
+    END IF;
+    --EXECUTE format('DROP TABLE IF EXISTS %I',output_table);
+    --EXECUTE format('CREATE TABLE %I(%I integer, %I DOUBLE PRECISION[], predlabel float)',output_table,id_column_name,test_column_name);
+	
+    EXECUTE
+        $sql$
+	DROP TABLE IF EXISTS pg_temp.madlib_knn_interm;
+	CREATE TABLE pg_temp.madlib_knn_interm AS    
+	select * from ( select row_number() over (partition by test_id order by dist) as r, x.* from (select test. $sql$ || id_column_name || $sql$ as test_id, madlib.squared_dist_norm2(train.$sql$ || point_column_name || $sql$,test.$sql$ || test_column_name || $sql$) as dist, $sql$ || label_column_name || $sql$ from $sql$ || textin(regclassout(point_source)) || $sql$ as train, $sql$ || textin(regclassout(test_source)) || $sql$ as test)x)y where y.r <= $sql$ || k;
+	IF (operation = 'c') THEN
```

Comment:
I was playing around with knn a bit, and observed that using `x` and `y` in `|| $sql$ as test)x)y where y.r` could cause some trouble if the column names corresponding to `label_column_name`, `point_column_name` or `test_column_name` are also named `x` or `y` (or even if there is another column called `y` in `test_source`). The query would fail to execute due to ambiguity.

In general, when you are using temp names in a sql query, its always best to using a unique string instead of something like x or y. MADlib has a python function named `unique_string()` in module `utilities.utilities` which will generate a unique string for you. I would suggest you apply this wherever you are using a temp name/placeholder in the SQL queries used in knn.